### PR TITLE
Entityless attributes for gui

### PR DESCRIPTION
**Pull request by DanoFecko**
extended attributes tab with options to edit entityless attirbutes

### CODE REVIEW

**Comment by DanoFecko** [comment]
@Vojtech-Sassmann I have made requested changes

File: perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+		if (!isAuthorized()) addButton.setEnabled(false);
+
+		// remove attr button
+		final CustomButton removeButton = TabMenu.getPredefinedButton(ButtonType.REMOVE, ButtonTranslation.INSTANCE.removeAttributes());
+		menu.addWidget(removeButton);
+		if (!isAuthorized()) removeButton.setEnabled(false);
+
+
+		// remove button event
+		final JsonCallbackEvents removeButtonEvent = JsonCallbackEvents.disableButtonEvents(removeButton, events);
```
Comment (Vojtech-Sassmann) [inline perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java]: What is this event for?

File: perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+	 * Title widget
+	 */
+	private Label titleWidget = new Label("Loading vo attributes");
+
+	//map of attributes and keys
+	private HashMap<Attribute, String> map;
+
+	private AttributeDefinition def;
+
+	public final static String URL = "settings";
```
Comment (Vojtech-Sassmann) [inline perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java]: 'settings' is not a very good name for this page.

File: perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+
+	@Override
+	public boolean multipleInstancesEnabled() {
+		return false;
+	}
+
+	@Override
+	public void open() {
+		session.getUiElements().getMenu().openMenu(MainMenu.PERUN_ADMIN);
+		session.getUiElements().getBreadcrumbs().setLocation(0, "Settings", getUrlWithParameters());
```
Comment (Vojtech-Sassmann) [inline perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java]: Same here.

File: perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributesDetailTabItem.java [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+import cz.metacentrum.perun.webgui.widgets.TabMenu;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Detail of Attribute definition
+ *
+ * @author Pavel Zlamal <256627@mail.muni.cz>
```
Comment (Vojtech-Sassmann) [inline perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributesDetailTabItem.java]: Correct this please.

File: perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java (hunk 1 above)
Comment (DanoFecko) [reply perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java]: it was remnant of me wanting to disable remove button when no attribute is selected. I removed it

File: perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java (hunk 2 above)
Comment (DanoFecko) [reply perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java]: changed it to edit_keys

**Comment by DanoFecko** [comment]
Hi @Vojtech-Sassmann, I made requested changes. To fix the bug with null showing, i created new JsonCallbackEvents and assigned them to JsonPostClient, where draw() method is called.  I hope this will work.

File: perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+
+	@Override
+	public boolean multipleInstancesEnabled() {
+		return false;
+	}
+
+	@Override
+	public void open() {
+		session.getUiElements().getMenu().openMenu(MainMenu.PERUN_ADMIN);
+		session.getUiElements().getBreadcrumbs().setLocation(0, "edit_keys", getUrlWithParameters());
```
Comment (Vojtech-Sassmann) [inline perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java]: This 'edit_keys' should be in a readable form. It is displayed in breadcrumb. I would use something like 'Entityless attributes setting'

Comment (DanoFecko) [reply perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java]: changed it to sugested 'Entityless attributes setting'

**Comment by DanoFecko** [comment]
@Vojtech-Sassmann I added option to edit first entry in map for new key

image

**Pull request merged**